src/rt/gtfsrt_update.cc

```diff
@@ -419,11 +432,102 @@ bool update_run(source_idx_t const src,
   return true;
 }
 
+void handle_vehicle_position(timetable const& tt,
+                             rt_timetable& rtt,
+                             source_idx_t const src,
+                             std::string_view tag,
+                             gtfsrt::FeedEntity const& entity,
+                             date::sys_days const today,
+                             date::sys_seconds const message_time,
+                             std::shared_ptr<opentelemetry::trace::Span>& span,
+                             statistics& stats) {
+
+  // handle accepted entities
+  try {
+    auto const& vp = entity.vehicle();
+    auto const vp_lat = vp.position().latitude();
+    auto const vp_lon = vp.position().longitude();
+    auto const& td = vp.trip();
+    auto const& gtfsrt_trip_id = td.trip_id();
+
+    auto [r, trip_idx] = gtfsrt_resolve_run(today, tt, &rtt, src, td,
+                                            std::string_view{gtfsrt_trip_id});
+    if (!r.valid()) {
+      log(log_lvl::error, "rt.gtfs.unsupported",
+          R"(Run could not be resolved. Skipping Message)", tag, entity.id());
+      return;
+    }
+
+    auto const& rtt_const = rtt;
+    auto const location_seq =
+        r.is_scheduled()
+            ? std::
+                  span{tt.route_location_seq_[tt.transport_route_[r.t_.t_idx_]]}
+            : std::span{rtt_const.rt_transport_location_seq_[r.rt_]};
+
+    // add rt_transport if not existent
+    if (!r.is_rt()) {
+      r.rt_ = rtt.add_rt_transport(src, tt, r.t_);
+    }
+
+    // match position to stop
+    auto const vp_position = geo::latlng{vp_lat, vp_lon};
+    auto const app_dist_lng_deg_vp =
+        geo::approx_distance_lng_degrees(vp_position);
+    auto const stop_it = utl::find_if(location_seq, [&](auto const& loc) {
+      auto const stop_position = tt.locations_.get(stop{loc}.location_idx());
+      return geo::approx_squared_distance(stop_position.pos_, vp_position,
+                                          app_dist_lng_deg_vp) < 10;
+    });
+    if (stop_it == end(location_seq)) {
+      log(log_lvl::info, "rt.gtfs.unsupported",
+          R"(Position of Vehicle was not near stop. Skipping Message)", tag,
+          entity.id());
+      ++stats.vehicle_position_position_not_at_stop_;
+      return;
+    }
+
+    // get remaining stops
+    auto const stopped_at_idx = stop_idx_t(std::distance(begin(location_seq), stop_it));
+    auto const fr = r.is_scheduled() ? frun::from_t(tt, &rtt_const, r.t_)
```

Comment:
there's no test for additional trips?
either skip them entirely for now or add a test

(update order matters: first vehicle position or first additional trip with tripUpdate)

Reply:
now is_scheduled is always true - why check?